mousestyles/data/__init__.py

```diff
@@ -62,3 +62,80 @@ def load_all_features():
     other_features = [x.iloc[:, -1] for x in data_frames]
     other_features.insert(0, all_features_df)
     return pd.concat(other_features, axis=1)
+
+
+def load_intervals(feature):
+    """
+    Parameters
+    ----------
+    feature: {"AS", "F", "IS", "M_AS", "M_IS", "W"}
+
+    Returns
+    -------
+    intervals : pandas.DataFrame
+        All the data from the feature directory into one data frame
+        columns are strain, mouse, day, start, stop
+    """
+    # get all file names
+    file_names = _os.listdir(_os.path.join(data_dir, "intervals", feature))
+    # initialized data frame
+    # use for loop to load every file and concat to overall data frame
+    dt = pd.DataFrame()
+    for item in file_names:
+        strain = int(item.split("strain")[1].split("_mouse")[0])
+        mouse = int(item.split("mouse")[1].split("_day")[0])
+        day = int(item.split("day")[1].split(".npy")[0])
+        path = _os.path.join(data_dir, "intervals", feature, item)
+        sub = np.load(path)
+        dt_sub = pd.DataFrame()
+        dt_sub["strain"] = [strain] * sub.shape[0]
+        dt_sub["mouse"] = [mouse] * sub.shape[0]
+        dt_sub["day"] = [day] * sub.shape[0]
+        dt_sub["start"] = sub[:, 0]
+        dt_sub["stop"] = sub[:, 1]
+        dt = pd.concat([dt, dt_sub])
+    # sort based on strain, mouse and day
+    dt = dt.sort(["strain", "mouse", "day"])
+    dt.index = range(dt.shape[0])
+    return dt
+
+
+def load_movement(strain, mouse, day):
+    """
+    Parameters
+    ----------
+    strain: int
+        nonnegative integer indicating the strain number
+    mouse: int
+        nonnegative integer indicating the mouse number
+    day: int
+        nonnegative integer indicating the day number
+
+    Returns
+    -------
+    movement : pandas.DataFrame
+        CT, CX, CY coordinates of the specified strain-mouse-day
+        columns are t, x, y and isHB
+        isHB indicates whether the point is in the home base or not
+    """
+    # load all four files of HB, CT, CX and CY data
+    HB_path = "txy_coords/C_idx_HB/C_idx_HB_strain{}_mouse{}_day{}.npy".\
+        format(strain, mouse, day)
+    CT_path = "txy_coords/CT/CT_strain{}_mouse{}_day{}.npy".\
+        format(strain, mouse, day)
+    CX_path = "txy_coords/CX/CX_strain{}_mouse{}_day{}.npy".\
+        format(strain, mouse, day)
+    CY_path = "txy_coords/CY/CY_strain{}_mouse{}_day{}.npy".\
+        format(strain, mouse, day)
+    HB = np.load(_os.path.join(data_dir, HB_path))
+    CT = np.load(_os.path.join(data_dir, CT_path))
+    CX = np.load(_os.path.join(data_dir, CX_path))
+    CY = np.load(_os.path.join(data_dir, CY_path))
+    # make data frame
+    dt = pd.DataFrame()
+    dt["t"] = CT
+    dt["x"] = CX
+    dt["y"] = CY
+    dt["isHB"] = HB
+    dt = dt.sort(["t"])
```

Comment:
Is the having the output Dataframe _sorted_ essential here? Can we remove this if unecesary. If sorting adds a trivial amount to the runtime (<1s) then you can probably leave it in.